phpunit/interactivity-api/class-wp-interactivity-api-directives-processor-test.php

```diff
@@ -366,4 +366,354 @@ public function test_is_void_element() {
+		$this->assertEquals( 'content-2', $p->get_attribute( 'class' ) ); // It didn't move.
+	}
+
+	/**
+	 * Tests the `append_content_after_closing_tag_on_balanced_or_void_tags`
+	 * method on a non-existent tag.
+	 *
+	 * @covers ::append_content_after_closing_tag_on_balanced_or_void_tags
+	 */
+	public function test_append_content_after_closing_tag_on_balanced_or_void_tags_non_existent_tag() {
+		$content_1 = 'Just a string with no tags.';
+		$content_2 = '<div>New text</div>';
+		$p         = new WP_Interactivity_API_Directives_Processor( $content_1 );
+		$p->next_tag();
+		$result = $p->append_content_after_closing_tag_on_balanced_or_void_tags( $content_2 );
+		$this->assertFalse( $result );
+		$this->assertEquals( $content_1, $p );
+	}
+
+	/**
+	 * Tests the `append_content_after_closing_tag_on_balanced_or_void_tags`
+	 * method with unbalanced tags.
+	 *
+	 * @covers ::append_content_after_closing_tag_on_balanced_or_void_tags
+	 */
+	public function test_append_content_after_closing_tag_on_balanced_or_void_tags_with_unbalanced_tags() {
+		$new_content = 'New text';
+
+		$content = '<div>Missing closing div';
+		$p       = new WP_Interactivity_API_Directives_Processor( $content );
+		$p->next_tag();
+		$result = $p->append_content_after_closing_tag_on_balanced_or_void_tags( $new_content );
+		$this->assertFalse( $result );
+		$this->assertEquals( $content, $p );
+
+		$content = '<div><div>Missing closing div</div>';
+		$p       = new WP_Interactivity_API_Directives_Processor( $content );
+		$p->next_tag();
+		$result = $p->append_content_after_closing_tag_on_balanced_or_void_tags( $new_content );
+		$this->assertFalse( $result );
+		$this->assertEquals( $content, $p );
+
+		$content = '<div>Missing closing div</span>';
+		$p       = new WP_Interactivity_API_Directives_Processor( $content );
+		$p->next_tag();
+		$result = $p->append_content_after_closing_tag_on_balanced_or_void_tags( $new_content );
+		$this->assertFalse( $result );
+		$this->assertEquals( $content, $p );
+
+		// It supports unbalanced tags inside the content, as long as it finds a
+		// balanced closing tag.
+		$content = '<div>Missing opening span</span></div>';
+		$p       = new WP_Interactivity_API_Directives_Processor( $content );
+		$p->next_tag();
+		$result = $p->append_content_after_closing_tag_on_balanced_or_void_tags( $new_content );
+		$this->assertTrue( $result );
+		$this->assertEquals( $content . $new_content, $p );
+	}
+
+	/**
+	 * Tests that the `next_balanced_tag_closer_tag` method finds a closing tag
+	 * for a standard tag.
+	 *
+	 * @covers ::next_balanced_tag_closer_tag
+	 */
+	public function test_next_balanced_tag_closer_tag_standard_tags() {
+			$content = '<div>Text</div>';
+			$p       = new WP_Interactivity_API_Directives_Processor( $content );
+			$p->next_tag();
+			$this->assertTrue( $p->next_balanced_tag_closer_tag() );
+			$this->assertEquals( 'DIV', $p->get_tag() );
+			$this->assertTrue( $p->is_tag_closer() );
```

Comment:
```suggestion
		$content = '<div>Text</div>';
		$p       = new WP_Interactivity_API_Directives_Processor( $content );
		$p->next_tag();
		$this->assertTrue( $p->next_balanced_tag_closer_tag() );
		$this->assertEquals( 'DIV', $p->get_tag() );
		$this->assertTrue( $p->is_tag_closer() );
```

It seems there's an extra tab in each line.

PS: Same in other tests below. 👇

Reply:
Thanks! Fixed in https://github.com/WordPress/gutenberg/pull/58498/commits/72a2f19de85622bfa4274aa5074776855c3b38c7.